src/d2_docker/utils.py

```diff
@@ -221,9 +223,12 @@ def run_docker_compose(
 
     env_pairs = [
         ("DHIS2_DATA_IMAGE", final_image_name),
-        ("DHIS2_CORE_PORT", str(port)) if port else None,
+        ("DHIS2_CORE_PORT", str(port) if port else None),
```

Comment:
This was a typo I guess

Reply:
It was intentional, the idea is to completely remove the pair if there is no port. But since we are both filtering `None` and `(key, None)`, both should work.